job/server/src/main/java/alluxio/job/transform/format/csv/CsvReader.java

```diff
@@ -39,14 +39,17 @@
 public final class CsvReader implements TableReader {
   private static final Logger LOG = LoggerFactory.getLogger(CsvReader.class);
 
+  private final FileSystem mFs;
   private final AvroCSVReader<Record> mReader;
   private final CsvSchema mSchema;
 
   /**
+   * @param fs the hdfs compatible client
    * @param reader the CSV reader
    * @param schema the schema
    */
-  private CsvReader(AvroCSVReader<Record> reader, Schema schema) {
+  private CsvReader(FileSystem fs, AvroCSVReader<Record> reader, Schema schema) {
```

Comment:
Would it make more sense to init the file system and reader in the `CsvReader` since they share a lifecycle?